tests/acceptance/test_mender-connect.py

```diff
@@ -0,0 +1,349 @@
+    def test_mender_connect_auth_changes(
+        self, connection,
+    ):
+        """Test that mender-connect can re-establish the connection on D-Bus signals"""
+
+        try:
+            proc_server_dbus = None
+            proc_server_ws_one = None
+            proc_server_ws_two = None
+
+            # start the mocks
+            put_mock_files(connection)
+            proc_server_dbus = start_mock_dbus_server(connection)
+            proc_server_ws_one = start_mock_websockets_server(connection, 5000)
+            proc_server_ws_two = start_mock_websockets_server(connection, 6000)
+
+            wait_for_mock_dbus_server(connection)
+            dbus_set_token_and_url(
+                connection, "whateversecuretoken", "http://localhost:5000"
+            )
+
+            # start the mender-connect service
+            startup_time = time.time()
+            connection.run(
+                "systemctl --job-mode=ignore-dependencies start mender-connect"
+            )
+
+            # wait for first connect
+            _ = wait_for_string_in_log(
+                connection,
+                startup_time,
+                30,
+                "Connection established with http://localhost:5000",
+            )
+
+            # 1. Change token
+            signal_time = time.time()
+            dbus_set_token_and_url_and_emit_signal(
+                connection, "newtoken", "http://localhost:5000"
+            )
+            _ = wait_for_string_in_log(
+                connection,
+                signal_time,
+                30,
+                "Connection established with http://localhost:5000",
+            )
+
+            # 2. Change url
+            signal_time = time.time()
+            dbus_set_token_and_url_and_emit_signal(
+                connection, "newtoken", "http://localhost:6000"
+            )
+            _ = wait_for_string_in_log(
+                connection,
+                signal_time,
+                30,
+                "Connection established with http://localhost:6000",
+            )
+
+            # 3. Change token and url
+            signal_time = time.time()
+            dbus_set_token_and_url_and_emit_signal(
+                connection, "newtokenagainverysecure", "http://localhost:5000"
+            )
+            _ = wait_for_string_in_log(
+                connection,
+                signal_time,
+                30,
+                "Connection established with http://localhost:5000",
+            )
+
+            # 4. Unauthorize and re-authorize
+            signal_time = time.time()
+            dbus_set_token_and_url_and_emit_signal(connection, "", "")
+            _ = wait_for_string_in_log(
+                connection,
+                signal_time,
+                30,
+                "dbusEventLoop terminated 0 sessions, 0 shells",
+            )
+            dbus_set_token_and_url_and_emit_signal(
+                connection, "reauthorizedtoken", "http://localhost:6000"
```

Comment:
Thoughts:
Seeing that previously the code was based on the length of the JWT token, maybe the test should use the same length tokens when changing them, to be sure?

Reply:
Good idea, I'll do that